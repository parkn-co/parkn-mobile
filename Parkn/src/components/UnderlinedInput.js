// @flow
import React, {Component, PropTypes} from 'react';
import {View, TextInput, Text, StyleSheet} from 'react-native';
import {bluePalette, grayPalette} from 'styles/colors';
import {fontNames} from 'styles/fonts';

export default class UnderlinedTextInput extends Component {
  state: {isActive: boolean};
  toggleActive: () => void;

  static propTypes = {
    label: PropTypes.string,
    onChangeText: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired,
    error: PropTypes.string,
  };

  constructor(props: Object) {
    super(props);

    this.state = {isActive: false};

    this.toggleActive = this.toggleActive.bind(this);
  }

  toggleActive() {
    this.setState({isActive: !this.state.isActive});
  }

  render(): React.Element<*> {
    const {label, error} = this.props;

    return (
      <View>
        <View style={[
          styles.container,
          this.state.isActive ? styles.active : {},
          this.props.error ? styles.withError : {},
        ]}>
          <Text style={styles.label}>{label}</Text>

          <TextInput
            style={styles.input}
            blurOnSubmit={true}
            onFocus={this.toggleActive}
            onBlur={this.toggleActive}
            selectionColor={grayPalette.white}
            {...this.props}
          />
        </View>
        {error && (
          <Text style={styles.error}>{error}</Text>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: 95,
    marginBottom: 38,
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(255,255,255,0.5)',
  },
  withError: {
    marginBottom: 5,
  },
  label: {
    color: grayPalette.white,
    opacity: 0.75,
    fontFamily: fontNames.light,
    fontSize: 20,
  },
  input: {
    flex: 1,
    fontFamily: fontNames.light,
    color: grayPalette.white,
    fontSize: 25,
    alignItems: 'flex-end',
    paddingBottom: 0,
  },
  error: {
    color: grayPalette.black,
    opacity: 0.75,
    fontFamily: fontNames.light,
    fontSize: 17,
    marginBottom: 10,
  },
  active: {
    borderBottomColor: grayPalette.white,
  }
});
