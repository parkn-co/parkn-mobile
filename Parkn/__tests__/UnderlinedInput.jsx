/* eslint-env jasmine */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import UnderlinedInput from '../src/components/UnderlinedInput';

it('renders UnderlinedInput component correctly', () => {
  expect(renderer.create(
    <UnderlinedInput onChangeText={() => null} value="mock value" />
  )).toMatchSnapshot();
});
