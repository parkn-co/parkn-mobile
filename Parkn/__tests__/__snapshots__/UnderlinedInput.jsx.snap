exports[`test renders UnderlinedInput component correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "rgba(255,255,255,0.5)",
          "borderBottomWidth": 1,
          "height": 95,
          "marginBottom": 38,
        },
        Object {},
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Light",
          "fontSize": 20,
          "opacity": 0.75,
        }
      } />
    <TextInput
      blurOnSubmit={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      selectionColor="#FFFFFF"
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#FFFFFF",
          "flex": 1,
          "fontFamily": "OpenSans-Light",
          "fontSize": 25,
          "paddingBottom": 0,
        }
      }
      value="mock value" />
  </View>
</View>
`;

exports[`test renders correctly 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "rgba(255,255,255,0.5)",
          "borderBottomWidth": 1,
          "height": 95,
          "marginBottom": 38,
        },
        Object {},
        Object {},
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "OpenSans-Light",
          "fontSize": 20,
          "opacity": 0.75,
        }
      } />
    <TextInput
      blurOnSubmit={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      selectionColor="#FFFFFF"
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#FFFFFF",
          "flex": 1,
          "fontFamily": "OpenSans-Light",
          "fontSize": 25,
          "paddingBottom": 0,
        }
      }
      value="mock value" />
  </View>
</View>
`;
